src/installation.js

```diff
@@ -0,0 +1,86 @@
+const fs = require('fs').promises
+const path = require('path')
+
+/* Creates symlinks recursively, sub-directories that already exist in
+ * `sourceDir` are not overwritten, they are accessed and symlinks are created
+ * inside them for the corresponding files in the `destDir` tree. If the
+ * symlink (file) to be created already exists in `sourceDir`, it is
+ * overwritten.
+ */
+const createSymlinks = async (sourceDir, destDir) => {
+  const filesCreated = []
+  const destFiles = await fs.readdir(destDir)
+
+  for (const destFile of destFiles) {
+    const sourceFilePath = path.join(sourceDir, destFile)
+    const destFilePath = path.join(destDir, destFile)
+
+    let sourceStats = null
+    try {
+      sourceStats = await fs.lstat(sourceFilePath)
+    } catch (error) {
+      if (error.code !== 'ENOENT') {
+        console.error(error)
+      }
+    }
+
+    if (sourceStats && sourceStats.isDirectory()) {
```

Comment:
shouldn't this be separated? Otherwise it'll jump into else block which will try to unlink `sourceFilePath` that got `sourceStats === null`. It will end up in `catch` so nothing bad will happen except for double logging the same error probably. Am I getting this right? 🤔

Reply:
you're right let me improve that, thanks